snuba/utils/metrics/__init__.py

```diff
@@ -0,0 +1,59 @@
+from random import random
+from collections import ChainMap
+from typing import Mapping, Optional, Union
+
+
+Tags = Mapping[str, str]
+
+
+from snuba.utils.metrics.backends.abstract import MetricsBackend
+
+
+class Metrics:
+    def __init__(
+        self,
+        backend: MetricsBackend,
+        namespace: Optional[str] = None,
+        tags: Optional[Tags] = None,
+        sample_rates: Optional[Mapping[str, float]] = None,
+    ) -> None:
+        self.__backend = backend
+        self.__namespace = namespace
+        self.__tags = tags
+        self.__sample_rates = sample_rates if sample_rates is not None else {}
+
+    def __apply_namespace(self, name: str) -> str:
+        if self.__namespace is not None:
+            return f"{self.__namespace}.{name}"
+        else:
+            return name
+
+    def __merge_tags(self, tags: Optional[Tags]) -> Tags:
+        return ChainMap(*[d for d in [self.__tags, tags] if d is not None])
```

Comment:
Since you always want to return something here, wouldn't it be easier to read if you initialized self.__tags = {}
and here just do:
> return ChainMap(self.__tags, tags or {})

Reply:
Good thinking, will make this change.